<!DOCTYPE html>
<head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">
    <title>PHINCH</title>
    <meta name="description" content="">
    <meta name="viewport" content="width=device-width">
    <link rel="stylesheet" href="css/utility.css">
    <link rel="stylesheet" href="css/preview.css">
    <script type="text/javascript">
        var _gaq = _gaq || [];
        _gaq.push(['_setAccount', 'UA-50346302-1']);
        _gaq.push(['_trackPageview']);
        (function() {
        var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
        ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
        var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
        })();
    </script>
</head>

<body>
    <section id = "head_sec">
        <p class = "logo"><a href="index.html"><img src="img/phinch-logo.jpg" height="50px"></a><img alt="Chrome" src="img/chrome_logo_2x.png" height="20px" style="margin-top: 20px;"></p>
        <h3>FILTER DATA</h3>
        <p id="help">HELP</p>
        <div id="dialog-modal">
            <h2>How to Use</h2>
            <p class="descPara">

                <span>Instructions</span></br></br>
                This visualization framework aims to address current bottlenecks in the analysis of large sequence datasets (rRNA amplicons, metagenomes), helping researchers analyze high-throughput datasets more efficiently. Phich takes advantage of standard outputs from computational pipelines in order to bridge the gap between biological software (e.g. QIIME) and existing data visualization capabilities (harnessing the scalability of WebGL and HTML5 in a browser-based tool). </br></br>
                Phinch currently supports downstream analyses of <b>.biom files</b> ( <a href="http://biom-format.org/" target="_blank">Biological Observation Matrix</a>, a JSON-formatted file type typically used to represent marker gene OTUs or metagenomic data). <b>All sample metadata and taxonomy/ontology information MUST be embedded in the .biom file before being uploaded into Phinch</b>.</br></br>
                In QIIME (version 1.7 or later), users can prepare the .biom file by executing the following commands:</br></br>
                First, construct an OTU table:</br></br>
                     <code>make_otu_table.py -i final_otu_map_mc2.txt -o otu_table_mc2_w_tax.biom -t rep_set_tax_assignments.txt</code></br></br>
                Where your input file (-i) is your OTU Map (defining clusters of raw sequences reads), and taxonomy file (-t) contains the taxonomy or gene ontology strings that correspond to each OTU.</br></br>
                Second, add your sample metadata to your .biom file:</br></br>
                    <code>add_metadata.py -i otu_table_mc2_w_tax.biom -o otu_table_mc2_w_tax_and_metadata.biom -m sample_metadata_mapping_file.txt</code></br></br>
                Where your input file (-i) is your .biom file from the previous step, and your mapping file (-m) is a tab-delimited file containing sample metadata (<a href="http://qiime.org/documentation/file_formats.html#metadata-mapping-files" target="_blank">formatted according to these QIIME instructions</a>).</br></br>
                After these two steps, you're ready to upload.</br></br>
                If you want to visualize biological data currently formatted as a tab-delimited text file (e.g. the style of OTU tables produced by older versions of QIIME), <a href="https://github.com/biom-format/biom-format/blob/df81277857a553e7e5c9679924e09861d8a5f61f/doc/documentation/biom_conversion.rst" target="_blank">please refer to this documentation for conversion instructions</a>. Phinch supports both "sparse" and "dense" BIOM formats (although <b>sparse .biom files are highly recommended, since the file size is much smaller</b>). </br></br>

                <span>Some important notes on metadata</span></br></br>
                In order to be properly detected, all date/time metadata must be standardized according to MIxS standardized format (<a href="http://wiki.gensc.org/index.php?title=MIxS" target="_blank">more information at the Genomic Standards Consortium wiki</a>), and entered into one column in your original sample metadata mapping file, as follows:</br></br>
                [YYYY]-[MM]-[DD]T[hh]:[mm]:[ss]-[Z]</br>
                For example, metadata for a sample collected at 2:30pm EST on May 4, 2007 would be entered as: 2007-04-05T14:30:00-05:00</br></br>
                Similarly, any geographic coordinates or GPS data must be entered as <a href="http://en.wikipedia.org/wiki/Decimal_degrees" target="_blank"></a>decimal degrees (the format used by GoogleMaps, e.g. -90.017926).  We recommend using separate columns labeled “Latitude” and “Longitude” in your original sample metadata mapping file, to ensure that GPS metadata is correctly detected.</br></br>

                To label your samples with human-readable IDs in the visualizations, include a column in your metadata mapping file with the header labelled as “phinch”. These IDs will be pulled through into the visualizations to populate graph axes. If this column is not included, an arbitrary numerical ID will be assigned to each sample.</br></br>
                                
                <span>Compatibility with older QIIME file formats</span></br></br>
                If you want to visualize biological data currently formatted as a tab-delimited text file (e.g. the style of OTU tables produced by older versions of QIIME), <a href="https://github.com/biom-format/biom-format/blob/df81277857a553e7e5c9679924e09861d8a5f61f/doc/documentation/biom_conversion.rst" target="_blank">please refer to this documentation for conversion instructions</a>. Phinch supports both "sparse" and "dense" BIOM formats (although <b>sparse .biom files are highly recommended, since the file size is much smaller</b>).</br></br>

            </p>            
        </div>
    </section>

    <section id = "up_sec">
        <p id = "file_details"></p>
        <button class="orange_btn" id="goGallery">PROCEED TO GALLERY</button>
        <button class="orange_btn" id="goExport">EXPORT</button>
    </section>

    <hr width="100%">
    </br>

    <section id = "NotSupported">
        <br /> <br />
        <h1>NOT SUPPORTED</h1>
        <p>The browser you are using does not support <a href="https://developer.mozilla.org/en-US/docs/IndexedDB" target="_blank">IndexedDB</a>, which is required for the Phinch framework!</p>
    </section>

    <section id = "bottom_sec">
        
        <section id = "left_control_panel">

            <p class = "att_head" id = "att_head_dates">Dates</p>
            <div id = "dates" class="att_block"></div></br>

            <p class = "att_head" id = "att_head_numeric">Numeric Range</p>
            <div id = "numeric_att" class="att_block"></div></br>
            
            <p class = "att_head" id = "att_head_groupable">Groupable</p>
            <div id = "groupable_att" class="att_block"></div></br>
            
            <p class = "att_head" id = "att_head_descriptive">Descriptive or Fixed Value</p>
            <div id = "non_numeric_att" class="att_block"></div>

        </section>

        <section id = "right_live_panel">
            <i class="icon-spinner icon-spin icon-large icon-4x" style="float:right;"></i>
            <!-- <div id = "myTable"></div> -->
        </section>

    </section>
    
    <script src="lib/jquery-1.10.1.min.js"></script>
    <script src="lib/jquery-ui-1.10.3.custom.min.js"></script>
    <script src="lib/indexedDB.polyfill.js"></script>
    <script src="lib/db.js"></script>
    <script src="lib/FileSaver.js"></script>
    <script src="lib/moment.min.js"></script>
    <script src="lib/d3.v3.min.js"></script>
    <script src="lib/jquery.dataTables.min.js"></script>

    <script src="scripts/filter.js"></script>

    <script>
    window.mobilecheck = function() {
        var check = false;
        (function(a){if(/(android|bb\d+|meego).+mobile|avantgo|bada\/|blackberry|blazer|compal|elaine|fennec|hiptop|iemobile|ip(hone|od)|iris|kindle|lge |maemo|midp|mmp|mobile.+firefox|netfront|opera m(ob|in)i|palm( os)?|phone|p(ixi|re)\/|plucker|pocket|psp|series(4|6)0|symbian|treo|up\.(browser|link)|vodafone|wap|windows (ce|phone)|xda|xiino/i.test(a)||/1207|6310|6590|3gso|4thp|50[1-6]i|770s|802s|a wa|abac|ac(er|oo|s\-)|ai(ko|rn)|al(av|ca|co)|amoi|an(ex|ny|yw)|aptu|ar(ch|go)|as(te|us)|attw|au(di|\-m|r |s )|avan|be(ck|ll|nq)|bi(lb|rd)|bl(ac|az)|br(e|v)w|bumb|bw\-(n|u)|c55\/|capi|ccwa|cdm\-|cell|chtm|cldc|cmd\-|co(mp|nd)|craw|da(it|ll|ng)|dbte|dc\-s|devi|dica|dmob|do(c|p)o|ds(12|\-d)|el(49|ai)|em(l2|ul)|er(ic|k0)|esl8|ez([4-7]0|os|wa|ze)|fetc|fly(\-|_)|g1 u|g560|gene|gf\-5|g\-mo|go(\.w|od)|gr(ad|un)|haie|hcit|hd\-(m|p|t)|hei\-|hi(pt|ta)|hp( i|ip)|hs\-c|ht(c(\-| |_|a|g|p|s|t)|tp)|hu(aw|tc)|i\-(20|go|ma)|i230|iac( |\-|\/)|ibro|idea|ig01|ikom|im1k|inno|ipaq|iris|ja(t|v)a|jbro|jemu|jigs|kddi|keji|kgt( |\/)|klon|kpt |kwc\-|kyo(c|k)|le(no|xi)|lg( g|\/(k|l|u)|50|54|\-[a-w])|libw|lynx|m1\-w|m3ga|m50\/|ma(te|ui|xo)|mc(01|21|ca)|m\-cr|me(rc|ri)|mi(o8|oa|ts)|mmef|mo(01|02|bi|de|do|t(\-| |o|v)|zz)|mt(50|p1|v )|mwbp|mywa|n10[0-2]|n20[2-3]|n30(0|2)|n50(0|2|5)|n7(0(0|1)|10)|ne((c|m)\-|on|tf|wf|wg|wt)|nok(6|i)|nzph|o2im|op(ti|wv)|oran|owg1|p800|pan(a|d|t)|pdxg|pg(13|\-([1-8]|c))|phil|pire|pl(ay|uc)|pn\-2|po(ck|rt|se)|prox|psio|pt\-g|qa\-a|qc(07|12|21|32|60|\-[2-7]|i\-)|qtek|r380|r600|raks|rim9|ro(ve|zo)|s55\/|sa(ge|ma|mm|ms|ny|va)|sc(01|h\-|oo|p\-)|sdk\/|se(c(\-|0|1)|47|mc|nd|ri)|sgh\-|shar|sie(\-|m)|sk\-0|sl(45|id)|sm(al|ar|b3|it|t5)|so(ft|ny)|sp(01|h\-|v\-|v )|sy(01|mb)|t2(18|50)|t6(00|10|18)|ta(gt|lk)|tcl\-|tdg\-|tel(i|m)|tim\-|t\-mo|to(pl|sh)|ts(70|m\-|m3|m5)|tx\-9|up(\.b|g1|si)|utst|v400|v750|veri|vi(rg|te)|vk(40|5[0-3]|\-v)|vm40|voda|vulc|vx(52|53|60|61|70|80|81|83|85|98)|w3c(\-| )|webc|whit|wi(g |nc|nw)|wmlb|wonu|x700|yas\-|your|zeto|zte\-/i.test(a.substr(0,4)))check = true})(navigator.userAgent||navigator.vendor||window.opera);
        return check; 
    }
    if (window.mobilecheck()) {
        $('body').css({'min-width': window.screen.width});
        $('#NotSupported p').html('Desktop Chrome Recommended!');
        $('#NotSupported').css({'width': window.screen.width, 'display': 'block'});
        $('h1, h3, #bottom_sec, #menu, #help, .orange_btn, hr').hide();
    } else if ( !navigator.userAgent.match(/Chrome/i) || !window.File || !window.FileReader || !window.FileList || !window.Blob ){
        // || navigator.userAgent.match(/Firefox/i) || navigator.userAgent.match(/Safari/) )
        $('#NotSupported').show();
        $('#head_sec').animate({height:50}, 200);
        $('#bottom_sec, h3, hr, .orange_btn').hide();
        alert('Please use Chrome or Firefox or Safari! Your browser does not support the Phinch framework!');
    } else {
        $('#help').click( function (e){
            e.preventDefault();
            $('#dialog-modal').show();
            $('#dialog-modal').dialog({ height: 140, modal: true });
            $('.ui-button-text').html('<i class="icon-remove icon-large"></i>');
        });
        app = new filter();
    }
    </script>

</body>
</html>

